### PR TITLE
CaddyFile for monolith Dendrite

### PR DESCRIPTION
**Pull request by ealib**
Sample [`CaddyFile`](https://caddyserver.com/docs/caddyfile) for Caddy 2.5.x for monolith Dendrite

This PR does not require testing because it just adds a sample configuration file.

* [x]  I have justified why this PR doesn't need tests.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/main/docs/CONTRIBUTING.md#sign-off)

Signed-off-by: `Emanuele Aliberti <dev@mtka.eu>`

### CODE REVIEW

**Review by neilalexander — APPROVED** [review]
Thanks for this!

**Review by genofire — COMMENTED** [review]
I do not unterstand, why this is needed. You have announce both server and client with your well-known to port 443 ...

File: docs/caddy/monolith/CaddyFile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	
+	handle /.well-known/matrix/client {
+		import matrix-well-known-header
+		respond `{ "m.homeserver": { "base_url": "https://matrix.example.com" } }` 200
+	}
+
+	import handle_errors_maintenance
+}
+
+example.com:8448 {
```
Comment (genofire) [inline docs/caddy/monolith/CaddyFile:50]: Why listen on 8448 ?

Comment (genofire) [reply docs/caddy/monolith/CaddyFile:50]: solved in #2579

**Pull request merged**